tree/ntuple/v7/test/ntuple_processor.cxx

```diff
@@ -165,11 +165,11 @@ TEST_F(RNTupleProcessorTest, ChainedChain)
 
 TEST_F(RNTupleProcessorTest, ChainedJoin)
 {
-   std::vector<RNTupleOpenSpec> ntuples{{fNTupleNames[0], fFileNames[0]}, {fNTupleNames[1], fFileNames[1]}};
-
    std::vector<std::unique_ptr<RNTupleProcessor>> innerProcs;
-   innerProcs.push_back(RNTupleProcessor::CreateJoin(ntuples, {}));
-   innerProcs.push_back(RNTupleProcessor::CreateJoin(ntuples, {"i"}));
+   innerProcs.push_back(
+      RNTupleProcessor::CreateJoin({fNTupleNames[0], fFileNames[0]}, {{fNTupleNames[1], fFileNames[1]}}, {}));
+   innerProcs.push_back(
+      RNTupleProcessor::CreateJoin({fNTupleNames[0], fFileNames[0]}, {{fNTupleNames[1], fFileNames[1]}}, {"i"}));
```

Comment:
Is this practically chaining together a join composition made of two files that are aligned with another join composition with unaligned events?

Reply:
Sort of but not really. Both joins use the same (aligned) RNTuples so they are aligned in both cases. However, in the second case the join table is used (i.e. we pretend they are unaligned). Tests for the truly unaligned case are covered already in `ntuple_processor_join.cxx`, this is more concerned with the composition itself.